doc/reference/trd-uart.md

```diff
@@ -0,0 +1,527 @@
+the original call and issue the originally scheduled callback
+(as if the call that `Err`'d with `BUSY` never happened). However,
+ it does not issue a callback for the call to `transmit_buffer`
+  that returned `Err`.
+
+The valid error codes for `transmit_buffer` are:
+  - `OFF`: the underlying hardware is not available, perhaps because it has
+    not been initialized or has been initialized into a different mode
+    (e.g., a USART has been configured to be a SPI).
+  - `BUSY`: the UART is already transmitting and has not made a transmission
+    complete callback yet.
+  - `SIZE`: `tx_len` is larger than the passed slice.
+  - `FAIL`: some other failure.
+
+Calling `transmit_buffer` while there is an outstanding
+`transmit_buffer` or `transmit_word` operation MUST return `Err(BUSY)`.
+
+The `TransmitClient::transmitted_buffer` callback indicates completion
+of a buffer transmission.  The `Result` indicates whether the buffer
+was successfully transmitted.  The `tx_len` argument specifies how
+many data words (defined by `Configure`) were transmitted. If the
+`rval` of `transmitted_buffer` is `Ok(())`, `tx_len` MUST be equal to
+the size of the transmission started by `transmit_buffer`, defined
+above.  A call to `transmit_word` or `transmit_buffer` made within
+this callback MUST NOT return `Err(BUSY)` unless it is because this is
+not the first call to one of these methods in the callback.  When this
+callback is made, the UART MUST be ready to receive another call. The
+valid `ErrorCode` values for `transmitted_buffer` are all of those
+returned by `transmit_buffer` plus:
+  - `CANCEL` if the call to `transmit_buffer` was cancelled by a call
+    to `abort` and the entire buffer was not transmitted.
+  - `SIZE` if the buffer could only be partially transmitted.
+
+3.2 `transmit_word` and `transmitted_word`
+===============================
+
+The `transmit_word` method transmits a single data word of data
+asynchronously.  The word length is determined by the UART
+configuration.  A UART implementation MAY choose to not implement
+`transmit_word` and `transmitted_word`.  There is a default
+implementation of `transmitted_word` so clients that do not use
+`receive_word` do not have to implement a callback.
+
+If `transmit_word` returns `Ok(())`, the implementation MUST call the
+`transmitted_word` callback in the future. If a call to
+`transmit_word` returns `Err`, the implementation MUST NOT issue a
+callback for this call, although if the it is `Err(BUSY)` is will
+issue a callback for the outstanding operation.  Valid `ErrorCode`
+results for `transmit_word` are:
+  - `OFF`: The underlying hardware is not available, perhaps because
+    it has not been initialized or in the case of a shared
+    hardware USART controller because it is set up for SPI.
+  - `BUSY`: the UART is already transmitting and has not made a
+    transmission callback yet.
+  - `NOSUPPORT`: the implementation does not support `transmit_word`
+    operations.
```

Comment:
```suggestion
```

I'm not sure we should allow this. If a chip really can't do this for some reason, then asking the implementing driver to statically allocate 1 word of memory and transparently convert this to a `transmit_buffer` internally is not a huge lift. I feel like runtime feature-failures like this should be avoided if we can.

Reply:
Take a look at how many implement it. ;)